src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -97,48 +103,43 @@ Point SpatialJoinAlgorithms::convertGeoPointToPoint(GeoPoint point) const {
 // ____________________________________________________________________________
 Id SpatialJoinAlgorithms::computeDist(const rtreeEntry& geo1,
                                       const rtreeEntry& geo2) const {
-  auto convertPoint = [&](AnyGeometry geometry, std::optional<Box> bbox) {
+  auto convertPoint = [&](const AnyGeometry& geometry,
+                          const std::optional<Box>& bbox) {
     Box areaBox;
-    if (bbox.has_value()) {
-      areaBox = bbox.value();
-    } else {
-      areaBox = boost::apply_visitor(BoundingBoxVisitor(), geometry);
-    }
+    areaBox =
+        bbox.value_or(boost::apply_visitor(BoundingBoxVisitor(), geometry));
```

Comment:
That syntax looks nice, unfortunately it always (even if not needed) calls the `apply_visitor`, and in that unneeded case also copies the bounding box (the first thing is probably worse).
If boxes are rather cheap to copy, then you can take again the `optional<Box>` by value, and then do
`if (bbox.has_value() { bbox = apply_visitor(...)}; const auto& areaBox = bbox.value(); ...`